# 
# PowerPC/Apus support patch
# Needed support for Amiga PowerUP boards.
# Author: mostly Roman Zippel <zippel@linux-m68k.org>
# Upstream status: got synced with 2.6.12 recently, so in better status than
# previous apus patches. Only 81k left.
# This is the part which needs checking and which is not yet applied, added for
# completeness though
#

diff -Nur -x CVS linux-2.6.13/include/video/vga.h linux-2.6.13-apus/include/video/vga.h
--- linux-2.6.13/include/video/vga.h	2005-08-29 01:41:01.000000000 +0200
+++ linux-2.6.13-apus/include/video/vga.h	2005-07-30 20:52:54.000000000 +0200
@@ -28,6 +28,7 @@
  * Ugh, we don't have PCI space, so map readb() and friends to use Zorro space
  * for MMIO accesses. This should make cirrusfb work again on Amiga
  */
+#include <asm/zorro.h>
 #undef inb_p
 #undef inw_p
 #undef outb_p
@@ -36,11 +37,17 @@
 #undef writeb
 #undef writew
 #define inb_p(port)	0
+#undef inw_p
 #define inw_p(port)	0
+#undef outb_p
 #define outb_p(port, val)	do { } while (0)
+#undef outw
 #define outw(port, val)		do { } while (0)
+#undef readb
 #define readb		z_readb
+#undef writeb
 #define writeb		z_writeb
+#undef writew
 #define writew		z_writew
 #endif
 #include <asm/byteorder.h>
